docs/source/architecture/contracts.md

```diff
@@ -0,0 +1,112 @@
+The parameters are:
+	* The address of the `MinersEscrow` contract
+	* The amount of staked tokens
+	* The periods for locking (which are serialized into an array of bytes)
+
+When staking tokens, the miner sets the number of periods the tokens will be locked, which must be no less than some minimal locking time (30 periods).
+In order to unlock tokens, the miner must be active during the time of locking (and confirm activity each period).
+Each stake is represented by the amount of tokens locked, and the stake's duration in periods.
+The miner can add a new stake using `MinersEscrow.deposit(uint256, uint256)` or `MinersEscrow.lock(uint256, uint256)` methods.
+The miner can split stake into two parts: one with the same duration and another with an extended duration.
+For this purpose, the `MinersEscrow.divideStake(uint256, uint256, uint256, uint256)` method is used.
+The first two parameters are used to identify the stake to divide and the last two for the extended part of the stake.
+When calculating locked tokens (`MinersEscrow.getLockedTokens(address, uint256)` method), all stakes that are active during the specified period are summed.
+
+
+### Ursula Confirms Activity
+
+In order to confirm activity every period, miners call `MinersEscrow.confirmActivity()` wherein activities for the next period are registered.
+The method `MinersEscrow.confirmActivity` is called every time the methods `MinersEscrow.deposit(uint256, uint256)` or `MinersEscrow.lock(uint256, uint256)` is called.
+The miner gets a reward for every confirmed period.
+
+### Ursula Generates Staking Rewards
+After the period of activity has passed, the miner may call `MinersEscrow.mint()` method which computes and transfers tokens to the miner's account.
+Also note that calls to `MinersEscrow.lock(uint256, uint256)` and `MinersEscrow.confirmActivity()` are included the `MinersEscrow.mint()` method.
+
+The reward value depends on the fraction of locked tokens for the period (only those who confirmed activity are accounted for).
+Also, the reward depends on the number of periods during which the tokens will be locked: if the tokens will be locked for half a year, the coefficient is 1.5.
+The minimum coefficient is 1 (when tokens will get unlocked in the next period), and the maximum is 2 (when the time is 1 year or more).
+The reward is calculated separately for each stake that is active during the mining period and all rewards are summed up.
+The order of calling `mint` by miners (e.g. who calls first, second etc) doesn't matter.
+Miners can claim their rewards by using the `witdraw(uint256)` method. Only non-locked tokens can be withdrawn.
+
+
+### Ursula Generates Policy Rewards
+Also the miner gets rewards for policies deployed.
+Computation of the reward happens every time `mint()` is called by the `updateReward(address, uint256)` method.
+In order to take the reward, the miner needs to call method `withdraw()` of the contract `PolicyManager`.
```

Comment:
Does the `withdraw` function here take a parameter?

Reply:
This `withdraw` function without parameters